executor/show_test.go

```diff
@@ -1001,3 +1002,28 @@ func (s *testSerialSuite1) TestShowCreateTableWithIntegerDisplayLengthWarnings(c
 		"  `e` bigint DEFAULT NULL\n" +
 		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
 }
+
+func (s *testSuite5) TestShowVar(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	var showSQL string
+	for _, v := range variable.SysVars {
+		if v.Scope != variable.ScopeNone && v.Scope != variable.ScopeGlobal|variable.ScopeSession {
+			if v.Scope == variable.ScopeGlobal {
+				showSQL = "show variables like '" + v.Name + "'"
+				res := tk.MustQuery(showSQL)
+				c.Check(res.Rows(), HasLen, 0)
```

Comment:
See that example, https://github.com/pingcap/tidb/issues/18517#issuecomment-657743046
The global-only variable should be shown in `show session variables`

Reply:
Handled.